processor/metricsgenerationprocessor/factory.go

```diff
@@ -0,0 +1,111 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricsgenerationprocessor
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "metricsgeneration"
+)
+
+var processorCapabilities = component.ProcessorCapabilities{MutatesConsumedData: true}
+
+// NewFactory returns a new factory for the Metrics Generation processor.
+func NewFactory() component.ProcessorFactory {
+	return processorhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		processorhelper.WithMetrics(createMetricsProcessor))
+}
+
+func createDefaultConfig() config.Processor {
+	return &Config{
+		ProcessorSettings: config.NewProcessorSettings(typeStr),
+	}
+}
+
+func createMetricsProcessor(
+	ctx context.Context,
+	params component.ProcessorCreateParams,
+	cfg config.Processor,
+	nextConsumer consumer.Metrics,
+) (component.MetricsProcessor, error) {
+	processorConfig := cfg.(*Config)
+	if err := validateConfiguration(processorConfig); err != nil {
+		return nil, err
+	}
+
+	metricsProcessor := newMetricsGenerationProcessor(buildInternalConfig(processorConfig), params.Logger)
+
+	return processorhelper.NewMetricsProcessor(
+		cfg,
+		nextConsumer,
+		metricsProcessor,
+		processorhelper.WithCapabilities(processorCapabilities))
+}
+
+// validateConfiguration validates the input configuration has all of the required fields for the processor
+// An error is returned if there are any invalid inputs.
+func validateConfiguration(config *Config) error {
+	for _, rule := range config.GenerationRules {
+		if rule.NewMetricName == "" {
+			return fmt.Errorf("missing required field %q", NewMetricFieldName)
+		}
+
+		if rule.Type == "" {
+			return fmt.Errorf("missing required field %q", GenerationTypeFieldName)
+		}
+
+		if rule.Type != "" && !rule.Type.isValid() {
```

Comment:
```suggestion
		if !rule.Type.isValid() {
```

`rule.Type` has already been validated as non-empty.